File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>hash;
        hash[0]=1;
        int count=0;
        int curr_sum=0;

        for(int num :nums){
            curr_sum +=num;
            if(hash.find(curr_sum-k)!=hash.end()){
                count+=hash[curr_sum-k];
            }
        

        hash[curr_sum]++;
        }
        return count;
    }
        };


        

        
    

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        m={0:1}
        current_sum=0
        count=0
        for i in range(len(nums)):
            current_sum=current_sum+nums[i]
            if current_sum-k in m :
                count+=m[current_sum-k]
            
            if current_sum in m:
                m[current_sum]+=1
            else:
                m[current_sum]=1
        return count




           

        